content/beginner/200_secrets/managing-sealing-key.md

```diff
@@ -0,0 +1,76 @@
+kubectl delete secret database-credentials -n octank
+kubectl delete sealedsecret database-credentials -n octank
+kubectl delete secret -n kube-system -l sealedsecrets.bitnami.com/sealed-secrets-key
+kubectl delete -f controller.yaml 
+```
+
+Now, put the Secret containing the private key back into the cluster using the **master.yaml** file.
+```
+kubectl apply -f master.yaml 
+kubectl get secret -n kube-system -l sealedsecrets.bitnami.com/sealed-secrets-key
+```
+Output:
+{{< output >}}
+NAME                      TYPE                DATA   AGE
+sealed-secrets-keyvk4pr   kubernetes.io/tls   2      23s
+{{< /output >}}
+
+
+Next, redeploy the SealedSecret CRD, controller and RBAC artifacts on your EKS.
+```
+kubectl apply -f controller.yaml
+kubectl get pods -n kube-system | grep sealed-secrets-controller
+```
+Output:
+{{< output >}}
+sealed-secrets-controller-84fcdcd5fd-gznc2            0/1     ContainerCreating   0          2s
+{{< /output >}}
+
+View the logs of the newly launched controller pod. Note that the name of the controller pod will be different in your cluster.
+```
+kubectl logs sealed-secrets-controller-84fcdcd5fd-ds5t6 -n kube-system
+```
+
+Output:
+{{< output >}}
+2020/03/07 23:07:15 Starting sealed-secrets controller version: v0.9.8
+2020/03/07 23:07:15 Searching for existing private keys
+2020/03/07 23:07:16 ----- sealed-secrets-keyvk4pr
+2020/03/07 23:07:16 HTTP server serving on :8080
+{{< /output >}}
+
+As you can see from the logs, the controller was able to find the existing Secret **sealed-secrets-keyvk4pr** in the *kube-system* namespace and therefore does not create a new key pair. Now, let's redeploy the SealedSecret and verify that the controller is able to successfully unseal it.
+```
+kubectl apply -f sealed-secret.yaml 
+kubectl logs sealed-secrets-controller-84fcdcd5fd-gznc2  -n kube-system
+```
+
+Output:
+{{< output >}}
+2020/03/07 23:07:15 Starting sealed-secrets controller version: v0.9.8
+2020/03/07 23:07:15 Searching for existing private keys
+2020/03/07 23:07:16 ----- sealed-secrets-keyvk4pr
+2020/03/07 23:07:16 HTTP server serving on :8080
+2020/03/07 23:07:54 Updating octank/database-credentials
+2020/03/07 23:07:54 Event(v1.ObjectReference{Kind:"SealedSecret", Namespace:"octank", Name:"database-credentials", UID:"7a27919c-60c8-11ea-a6b7-0e57dc790b09", APIVersion:"bitnami.com/v1alpha1", ResourceVersion:"6409442", FieldPath:""}): type: 'Normal' reason: 'Unsealed' SealedSecret unsealed successfully
+{{< /output >}}
+
+{{% notice info %}}
+If file **master.yaml** which contains the public/private key pair generated by the controller is compromised, then all the SealedSecret manifests can be unsealed and the encrypted sensitive information they store revealed. Hence, this file must be guarded by granting least privilege access. Fo additional guidance on sealing key renewal, manual sealing key management etc., please consult the [documentation](https://github.com/bitnami-labs/sealed-secrets#secret-rotation).
```

Comment:
Typo - For

Reply:
Since master.yaml holds the 'keys to the kingdom', could you expand or provide a section on how should the reader do this? AWS Secrets Manager with strict IAM policies might be a good alternative.

Reply:
Yes, good point. I added additional notes about storing it as a SecureString in SSM Parameter Store using the Advanced tier.